core/src/block_sync.rs

```diff
@@ -193,21 +200,34 @@ pub mod message {
                         warn!("Error: not sending any blocks as batch_size is equal to zero.");
                         return;
                     }
-                    if *latest_hash == block_sync.sumeragi.latest_block_hash() {
+                    let local_latest_block_hash =
+                        block_sync.sumeragi.wsv_mutex_access().latest_block_hash();
+                    if *latest_hash == local_latest_block_hash
+                        && *previous_hash == local_latest_block_hash
```

Comment:
This condition will never be true (except case when `latest_hash` == `previous_hash`).
Should be `||` i think.